libs/animation/animation.ts

```diff
@@ -279,25 +281,37 @@ namespace animation {
 
         public add(node: PathNode) {
             this.nodes.push(node);
+            this.nodesEnabled.push(true);
         }
 
         get length(): number {
             return this.nodes.length;
         }
 
+        public resetNodes() {
+            this.nodesEnabled = [];
+            for (let i of this.nodes) {
+                this.nodesEnabled.push(true);
+            }
+        }
+
         public run(interval: number, target: Sprite, startedAt: number): boolean {
             const runningTime = control.millis() - startedAt; // The time since the start of the path
             const nodeIndex = Math.floor(runningTime / interval); // The current node
             const nodeTime = runningTime % interval; // The time the current node has been animating
 
             if (this.lastNode > -1 && this.lastNode < nodeIndex && this.nodes.length) { // If the last node hasn't been completed yet
-                this.nodes[this.lastNode].apply(target, interval, interval); // Applies the last state of the previous node in case it was missed (this makes sure all moveTos fire)
-
+                if (this.nodesEnabled[this.lastNode]) this.nodes[this.lastNode].apply(target, interval, interval); // Applies the last state of the previous node in case it was missed (this makes sure all moveTos fire)
                 if (nodeIndex >= this.nodes.length) return true; // Once the nodeIndex is past the last item of the array, only then end the animation
             }
-            this.lastNode = nodeIndex;
 
-            this.nodes[nodeIndex].apply(target, nodeTime, interval);
+            this.lastNode = nodeIndex;
+            if (this.nodesEnabled[nodeIndex]) {
```

Comment:
Not sure that nodesEnabled needs to be an array. You only check it for the current node, right?

Reply:
only for current node and the node before. I can switch it to be two bools instead!